src/app/components/weather/weather.controller.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+export default class WeatherController {
+    constructor(weatherAPIService, geolocationService, localStorageService) {
+        'ngInject';
+        
+        this.weatherAPIService = weatherAPIService;
+        this.geolocationService = geolocationService;
+        this.localStorageService = localStorageService;
+
+        this.weatherData = {};
+        this.selectedLocation = null;
+        this.tableShow = false;
+    }
+
+    getForecastForSavedLocation() {
+      const coords = this.localStorageService.getCoordinates();
+      
+      if (coords) {
+        this.weatherAPIService.getForecast(coords)
+        .then(data => {
+          this.localStorageService.setCoordinates(coords);
+          this.weatherData = data;
+          this.tableShow = true;
+        });
+      }
+    }
+
+    getForecastForSelected() {
+      const location = this.autocomplete.details.geometry.location;
+
+      const coords = {
+        lat: location.lat(),
+        long: location.lng()
+      }
+      
+      this.weatherAPIService.getForecast(coords)
+        .then(data => {
+          this.localStorageService.setCoordinates(coords);
+          this.weatherData = data;
+          this.tableShow = true;
```

Comment:
Why? The table with forecast data has this attribute `ng-show="weatherCtrl.tableShow"`, because the table is hidden when this one is empty or data is unavailable.

Reply:
Done ✔